### PR TITLE
Enabling test for 2447, CancellationToken does not gets validated.

### PR DESCRIPTION
**Pull request by harshgMSFT**

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ private class Person
             public CancellationToken Token { get; set; }
         }
 
-        [Fact(Skip = "CancellationToken should not be validated #2447.")]
+        [Fact]
         public async Task BindProperty_WithData__WithPrefix_GetsBound()
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs]: extra `_` here

File: test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs [hunk 2]
```diff
@@ -55,18 +55,10 @@ private class Person
             // ModelState
             Assert.True(modelState.IsValid);
 
-            Assert.Equal(2, modelState.Keys.Count);
-            Assert.Single(modelState.Keys, k => k == "CustomParameter");
-
-            var key = Assert.Single(modelState.Keys, k => k == "CustomParameter.Token");
-            Assert.Null(modelState[key].Value);
-            Assert.Empty(modelState[key].Errors);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs]: Assuming this part was affected by an earlier change an not just updated?

File: test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs (hunk 1 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs]: Also a few others in this file

**Comment by rynowak** [comment]
:shipit:

File: test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs (hunk 2 above)
Comment (harshgMSFT) [inline test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs]: yes.

File: test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs [hunk 3]
```diff
@@ -140,13 +126,7 @@ private class Person
...
             // ModelState
             Assert.True(modelState.IsValid);
-            var key = Assert.Single(modelState.Keys);
-            Assert.Equal("CustomParameter", key);
-            Assert.Null(modelState[key].Value);
-            Assert.Empty(modelState[key].Errors);
-
-            // This assert fails.
-            Assert.Equal(ModelValidationState.Skipped, modelState[key].ValidationState);
+            Assert.Equal(0, modelState.Count);
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.IntegrationTests/CancellationTokenModelBinderIntegrationTest.cs]: prefer `Assert.Empty(modelState);` but that's minor

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**